src/mesh/boundary_info.C

```diff
@@ -1357,8 +1364,8 @@ void BoundaryInfo::remove_edge (const Elem * elem,
 {
   libmesh_assert(elem);
 
-  // Only level 0 elements are stored in BoundaryInfo.
-  libmesh_assert_equal_to (elem->level(), 0);
+  // Only level 0 elements unless the flag "_children_on_boundary" is on.
+  libmesh_assert(elem->level()==0 || _children_on_boundary);
```

Comment:
Does `_children_on_boundary` get set by `add_edge` yet?

Reply:
No.

Reply:
BTW, what is difference between add_edge and add_side?

Reply:
In 2D nobody uses `add_edge`.  In 3D, `add_edge` associates a boundary id with just one 1D edge of a polyhedron, not a whole 2D side.